test/unit/image_data_test.rb

```diff
@@ -30,6 +30,41 @@ class ImageDataTest < ActiveSupport::TestCase
     assert image_data.valid?
   end
 
+  test "rejects images with duplicate filename on edition" do
+    image_data = build_example("960x640_jpeg.jpg")
+    images = [create(:image, image_data:)]
+    edition = create(:news_article, images:)
+    image = create(:image, image_data:)
+
+    image_data.validate_on_edition = edition
+    image_data.validate_on_image = image
```

Comment:
Make sure the new `image` is associated with the existing `edition`. (Assuming you remove `validate_on_edition`)

```suggestion
    image_data = build_example("960x640_jpeg.jpg")
    edition = create(:news_article, images: [build(:image, image_data:)])
    image = build(:image, edition:, image_data:)

    image_data.validate_on_image = image
```